app/src/main/java/com/islaharper/dawnofandroid/presentation/components/RatingWidget.kt

```diff
@@ -0,0 +1,353 @@
+package com.islaharper.dawnofandroid.presentation.components
+
+import android.content.res.Configuration.UI_MODE_NIGHT_YES
+import android.util.Log
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.drawscope.clipPath
+import androidx.compose.ui.graphics.drawscope.scale
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.graphics.vector.PathParser
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.ui.theme.DarkStarBackgroundColour
+import com.islaharper.dawnofandroid.ui.theme.DarkStarColour
+import com.islaharper.dawnofandroid.ui.theme.DawnOfAndroidTheme
+import com.islaharper.dawnofandroid.ui.theme.LightStarBackgroundColour
+import com.islaharper.dawnofandroid.ui.theme.LightStarColour
+import com.islaharper.dawnofandroid.ui.theme.PADDING_SMALL
+import com.islaharper.dawnofandroid.util.Constants.STAR_PATH
+
+@Composable
+fun RatingWidget(
```

Comment:
Thanks for this. Is this based on a sample somewhere e.g., in a course or in an article or Stack Overflow? If it is, please add a link or PDF if paywalled. This is really important to help reviewers understand the full context.

Reply:
Hi, I looked at https://www.jetpackcompose.app/snippets/RatingBar as a starting point and implemented something I was happy with.  I implemented a similar solution in the FNB App for a rating widget using the path from an SVG a while back. Don't have an example of previous java implementation as it's in the FNB codebase but I've attached a screenshot from it's current use in the FNB app.

![rating](https://github.com/DurtDiv3r/Dawn-of-Android/assets/75084231/16ed9db1-44bd-42b8-b1bc-a61f4884c042)